pkg/multisort/multisorter.go

```diff
@@ -47,18 +47,18 @@ func (s *Sorter[E]) SortBy(data []E, keys ...Key) error {
 		return err
 	}
 
-	slices.SortStableFunc(data, func(a, b E) bool {
+	slices.SortStableFunc(data, func(a, b E) int {
 		for _, key := range keys {
 			f := s.fields[key.ID]
 
 			switch f(a, b, key.Descending) {
 			case Less:
-				return true
+				return -1
 			case NotEqual:
-				return false
+				return 1
 			}
 		}
-		return false
+		return -1
```

Comment:
According to the docs of `SortFunc`:

```go
// SortFunc sorts the slice x in ascending order as determined by the cmp
// function. This sort is not guaranteed to be stable.
// cmp(a, b) should return a negative number when a < b, a positive number when
// a > b and zero when a == b.
//
// SortFunc requires that cmp is a strict weak ordering.
// See https://en.wikipedia.org/wiki/Weak_ordering#Strict_weak_orderings.
```

So in case the all fields are equal, we need to return `0`.

```suggestion
		return 0
```